sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/utils/SchemaUtil.scala

```diff
@@ -213,13 +238,21 @@ object SchemaUtil {
 
       stateVarType match {
         case ValueState =>
-          Seq("key", "single_value", "partition_id")
+          Seq("key", "value", "partition_id")
 
         case ListState =>
-          Seq("key", "list_value", "partition_id")
+          if (sourceOptions.flattenCollectionTypes) {
+            Seq("key", "value", "partition_id")
```

Comment:
Would we like to differentiate this with ValueState? like `list_element`. Users would be able to expect that there will be multiple values for the key based on the schema.

I'm also OK to assume that users would know about the state type for the variable. Just a 2 cents, as we are doing that for other cases (`list_value`, `user_map_key/value`, `map_value`).

Reply:
Done